daemon/cmd/daemon_main.go

```diff
@@ -1716,16 +1724,45 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], promise.Pr
 			daemon = d
 
 			if !option.Config.DryMode {
-				if err := startDaemon(daemon, restoredEndpoints, cleaner, params); err != nil {
-					daemonResolver.Reject(err)
-					cancelDaemonCtx()
-					cleaner.Clean()
-					cfgResolver.Reject(err)
-					return err
+				log.Info("Initializing daemon")
+
+				// This validation needs to be done outside of the agent until
+				// datapath.NodeAddressing is used consistently across the code base.
+				log.Info("Validating configured node address ranges")
+				if err := node.ValidatePostInit(); err != nil {
+					return fmt.Errorf("postinit failed: %w", err)
+				}
+
+				// Store config in file before resolving the DaemonConfig promise.
+				err = option.Config.StoreInFile(option.Config.StateDir)
+				if err != nil {
+					log.WithError(err).Error("Unable to store Cilium's configuration")
+				}
+
+				err = option.StoreViperInFile(option.Config.StateDir)
+				if err != nil {
+					log.WithError(err).Error("Unable to store Viper's configuration")
 				}
 			}
-			daemonResolver.Resolve(daemon)
+
+			// 'option.Config' is assumed to be stable at this point, execpt for
```

Comment:
This isn't necessarily true, startDaemon does performs many routines (most of which related to KPR configuration) that may override vars in `*option.DeamonConfig`.  The intention of this promise was a way to wait for specifically those flags.

[Here's ](https://github.com/cilium/cilium/pull/32558/files#diff-8a383aefc352ba8854dbab293a563fecf7f8916983d013ab36f677293996ca3cR1359-R1397) a list from a draft PR I was working on of all the configs that can change during the execution of `startDaemon`.

Reply:
Examples of where startDaemon is required is the `finishKubeProxyReplacementInit` which can disable `EnableNodePort` (which currently must be in startDaemon because it waits for device detection to complete).

Reply:
It looks to me like `finishKubeProxyReplacementInit()` as well as `disableNodePort()` calls are made in `newDaemon()` and not in `startDaemon()`. So these would NOT be examples of config changing after the DaemonConfig promise is resolved.